includes/data-stores/class-wc-product-variation-data-store-cpt.php

```diff
@@ -139,13 +143,17 @@ public function create( &$product ) {
 	 */
 	public function update( &$product ) {
 		$product->save_meta_data();
-		$changes = $product->get_changes();
-		$title   = $this->generate_product_title( $product );
+		$changes   = $product->get_changes();
```

Comment:
I think we should probably get the changes after we check/set the product name. Otherwise, it won't show up in the 'changes' array.